sdk/openai/openai-assistants/README.md

```diff
@@ -0,0 +1,334 @@
+# Azure Assistants REST client library for JavaScript
+
+Azure OpenAI APIs for Assistants.
+
+**Please rely heavily on our [REST client docs](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/rest-clients.md) to use this library**
+
+Key links:
+
+- [Package (NPM)](https://www.npmjs.com/package/@azure/openai-assistants)
+- [API reference documentation](https://docs.microsoft.com/javascript/api/@azure/openai-assistants)
+
+## Getting started
+
+### Currently supported environments
+
+- LTS versions of Node.js
+
+### Prerequisites
+
+- You must have an [Azure subscription](https://azure.microsoft.com/free/) to use this package.
+
+### Install the `@azure/openai-assistants` package
+
+Install the Azure Assistants REST client REST client library for JavaScript with `npm`:
+
+```bash
+npm install @azure/openai-assistants
+```
+
+### Create and authenticate a `AssistantsClient`
+
+To use an [Azure Active Directory (AAD) token credential](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/samples/AzureIdentityExamples.md#authenticating-with-a-pre-fetched-access-token),
+provide an instance of the desired credential type obtained from the
+[@azure/identity](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#credentials) library.
+
+To authenticate with AAD, you must first `npm` install [`@azure/identity`](https://www.npmjs.com/package/@azure/identity) 
+
+After setup, you can choose which type of [credential](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#credentials) from `@azure/identity` to use.
+As an example, [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#defaultazurecredential)
+can be used to authenticate the client.
+
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables:
+AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET
+
+
+## Usage
+
+### Overview
+
+See [OpenAI's "how assistants work"](https://platform.openai.com/docs/assistants/how-it-works) documentation for an
+overview of the concepts and relationships used with assistants. This overview closely follows
+[OpenAI's overview example](https://platform.openai.com/docs/assistants/overview) to demonstrate the basics of
+creating, running, and using assistants and threads.
+
+To get started, create an `AssistantsClient`:
+```javascript Snippet:OverviewCreateClient
+const assistantsClient = isAzureOpenAI
+  ? new AssistantsClient(new Uri(azureResourceUrl), new AzureKeyCredential(azureApiKey))
+  : new AssistantsClient(new OpenAIKeyCredential(nonAzureKey));
+```
+
+With a client, an assistant can then be created:
```

Comment:
Should we go into a little more detail about what an assistant is? More generally, we can add a description for every concept listed below as we introduce them.

Reply:
[Updated](https://github.com/Azure/azure-sdk-for-js/pull/27807/commits/d48c588a67f67d6fa937476a5c82750bd3e9afe4)